crates/sui-json-rpc/src/name_service.rs

```diff
@@ -224,3 +241,101 @@ pub struct NameRecord {
     /// Additional data which may be stored in a record
     pub data: VecMap<String, String>,
 }
+
+impl NameRecord {
+    /// Parses the name record's metadata and returns the `objectId` of the registry table.
+    pub fn namespace_table_id(&self) -> Option<ObjectID> {
+        let table_id_entry = self
+            .data
+            .contents
+            .iter()
+            .find(|entry| entry.key == NAMESPACE_TABLE_KEY)?;
+
+        ObjectID::from_str(&table_id_entry.value).ok()
+    }
+
+    /// Checks if a name record has expired.
+    pub fn is_expired(&self) -> bool {
```

Comment:
This should be a property that everyone in the world agrees on, but here we're depending on the system time of the RPC node that we're asking the question on -- that doesn't seem right.  Would recommend you use a checkpoint timestamp for this, so it's consistent no matter who's asking.

Reply:
I indeed need more education here, wasn't sure where to get a "trusted" UTC timestamp, so I assumed we trust the machine. Will look it up, thanks!